lib/std/heap.zig

```diff
@@ -417,7 +417,7 @@ pub const FixedBufferAllocator = struct {
     fn alloc(ctx: *anyopaque, n: usize, log2_ptr_align: u8, ra: usize) ?[*]u8 {
         const self: *FixedBufferAllocator = @ptrCast(@alignCast(ctx));
         _ = ra;
-        const ptr_align = @as(usize, 1) << @as(Allocator.Log2Align, @intCast(log2_ptr_align));
+        const ptr_align = if (@inComptime()) 1 else @as(usize, 1) << @as(Allocator.Log2Align, @intCast(log2_ptr_align));
```

Comment:
this is incorrect alignment. why would it be OK to do this?

Reply:
is alignment meaningful for comptime memory? this does seem pretty sus, but it seems to work.

Reply:
Comptime memory doesn't currently care about alignment, but that could change: in general, comptime tries to be strict in the sense of making code that would be UB at runtime a compile error.

The actual issue here is that `std.mem.alignPointerOffset` is trying to use `@intFromPtr` to align the pointer if it has the potential to be under-aligned. The problem is, without being able to use concrete addresses, we can't align to a boundary any higher than the FBA buffer's alignment.

I'm not sure what the correct fix is here, but as a stopgap solution to make things work, I think it'd be reasonable to merge this PR. However, we should make an issue to track this open design question and add a comment here referencing it.